src/render/tutorials/caching-http-requests/2_0.html.md

```diff
@@ -0,0 +1,319 @@
+If you want to learn more about tasks please check [knotx-fragments documentation](https://github.com/Knotx/knotx-fragments) 
+
+Let's start with defining a new task: `knotx/conf/routes/handlers/fragments-handler.conf`
+
+```hocon
+tasks {
+  api-task {
+    action = fetch-product-with-cache
+    onTransitions {
+      _success {
+        action = product-to-body
+      }
+    }
+  }
+}
+
+actions {
+  fetch-product-with-cache {
+    factory = in-memory-cache
+    config {
+      cache {
+        maximumSize = 1000
+        ttl = 5000
+      }
+      cacheKey = "product-{param.id}"
+      payloadKey = fetch-product
+    }
+    doAction = fetch-product
+  }
+
+  product-to-body {
+    factory = payload-to-body
+    config {
+      key = fetch-product
+    }
+  }
+
+  fetch-product {
+    factory = http
+    config {
+      endpointOptions {
+        path = /product/id
+        domain = webapi
+        port = 8080
+        allowedRequestHeaders = ["Content-Type"]
+      }
+    }
+  }
+}
+```
+As you can see above, to achieve our goal we have to create only one task called `api-task` - this is very simple directed graph.
+The action it performs is `fetch-product-with-cache` which we'll create soon. Then we have `onTransitions`, in our case let's 
+define only `_success` - the action to be performed is `product-to-body`  which we'll also define.
+
+Below task definition you can see all necessary action's definitions. As mentioned before, we've created `fetch-product-with-cache`. The purpose 
+of this action is to add cache functionality. Now you can see how easily we can define custom behaviours for our actions. 
+We've just assigned `in-memory-cache` action factory to it Moreover this action also wraps another one - in this case it's 
+`fetch-product`. It's using simple http factory which will call desired endpoint, in our case `/product/id` (we'll define it later).
+Finally we can focus on `product-to-body` - it's just adding previously fetched data to response body.
+
+## Why do we need caching?
+We've just added caching behaviour to action but you may wonder what are the benefits? Why we're using cache? The general purpose of
+using caching is faster data retrieval - when the user is calling the same endpoint a few times, we don't have to fetch the data
+from the server for every request - we can cache and reuse the response in the future (if not changed) - it effects in a quicker
+response time, which improves the overall user experience. Other benefits include reduced latency and traffic. Requests
+are satisfied in a shorter time with cached values.
+
+You may also wonder if we can replace this cache with another one, for example with [Redis](https://redis.io/). The answer is yes, you can do it very simply!
+The only thing needed here is to create appropriate custom action factory which will handle caching in the desired way and then properly link it to an action
```

Comment:
I thought we wanted to provide an example of redia implementation with Knot.x?

Reply:
probably we'll provide it in the future